### PR TITLE
sort module types alphabetically

### PR DESCRIPTION
**Pull request by danielneis**
this sorts module types alphabetically making it easier to find them on the list
hope you like

best,
Daniel

### CODE REVIEW

**Comment by mudrd8mz** [comment]
Thank you @danielneis for contributing this! I am never sure which solution is a good one for sorting plugin types. Sometimes (such as in the Moodle administration UI itself) it feels more natural when more essential plugin types (such as activity modules) are enlisted first as they are simply more important. And then it makes sense to have subplugins below their parent type etc.

However I agree that in this case, alphabetical order is probably more appropriate.

In either case, as the values here are localised strings, we will need UTF-8 safe sorting, not the PHP native `asort()`. Please amend the patch and use https://docs.moodle.org/dev/String_API#asort.28.29

File: classes/step0_form.php [hunk 1]
```diff
@@ -46,6 +46,7 @@ public function definition () {
         $mform->setExpanded('manualhdr', true);
 
         $plugintypes = tool_pluginskel\local\util\manager::get_plugintype_names();
+        asort($plugintypes);
```
Comment (jrchamp) [inline classes/step0_form.php:49]: Is this what you mean? I wonder if there are places in core Moodle that could also benefit from this change.
```suggestion
        \core_collator::asort($plugintypes);
```

Comment (mudrd8mz) [reply classes/step0_form.php:49]: > Is this what you mean?

Yup, that.

> in core Moodle

See my comment above - this was discussed before and there are pros and cons.

Comment (jrchamp) [reply classes/step0_form.php:49]: > > in core Moodle
> 
> See my comment above - this was discussed before and there are pros and cons.

Of course! I was not recommending that we "always sort alphabetically", but that places where raw `asort()` is already being used on arrays whose values are user-facing strings may want to consider the core_collator's asort() instead. Example: https://github.com/moodle/moodle/blob/511a87f5fc357f18a4c53911f6e6c7f7b526246e/mod/lesson/editpage_form.php#L49

Comment (mudrd8mz) [reply classes/step0_form.php:49]: Oh yes of course. Well spotted!

**Comment by danielneis** [comment]
thank you, David and Jonathan!

**Comment by mudrd8mz** [comment]
Thanks both. To be released as a part of 1.5.2

**Pull request closed without merging**